tests/tensorboard_test.py

```diff
@@ -0,0 +1,233 @@
+        path, event_file_loader.EventFileLoader).Load()
+    event_values = itertools.chain.from_iterable(
+        map(_process_event, event_file_generator))
+    for value_dict in event_values:
+      event_value_list.append(value_dict)
+
+    self.assertLen(event_value_list, 1)
+    self.assertEqual(event_value_list[0]['step'], 1)
+    self.assertGreater(event_value_list[0]['wall_time'], 0.0)
+    return event_value_list[0]['value']
+
+  def test_summarywriter_flush_after_close(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    summary_writer.close()
+    with self.assertRaises(AttributeError):
+      summary_writer.flush()
+
+  def test_summarywriter_scalar(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    # Write the scalar and check if the event exists and check data.
+    float_value = 99.1232
+    summary_writer.scalar(tag='scalar_test', value=float_value, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'scalar_test')
+    self.assertTrue(onp.allclose(
+        tensor_util.make_ndarray(summary_value.tensor).item(),
+        float_value))
+
+  def test_summarywriter_text(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    text = 'hello world.'
+    summary_writer.text(tag='text_test', textdata=text, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'text_test')
+    self.assertEqual(
+        tensor_util.make_ndarray(summary_value.tensor).item().decode('utf-8'),
+        text)
+
+  def test_summarywriter_image(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=255., size=(30, 30, 3))
+    img = img.astype(onp.uint8)
+    summary_writer.image(tag='image_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, 'image_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    self.assertTrue(onp.allclose(img, expected_img.numpy()))
+
+  def test_summarywriter_image_float_pixel_values(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=1., size=(30, 30, 3))
+    summary_writer.image(tag='image_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, 'image_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # convert img to uint8 as the decoded image is of type uint8.
+    actual_img = tf.image.convert_image_dtype(image=img, dtype=onp.uint8)
+    self.assertTrue(onp.allclose(actual_img, expected_img.numpy()))
```

Comment:
`actual_img` is a TF tensor, isn't it? Shouldn't there be a `.numpy()` for it too?

Reply:
Since expected_img is a numpy of dtype float32 and in the range of 0.0 to 1.0, summary writer internally scales up the pixel values accordingly. So here we need to do the same for assertion. So we scale up the values and also set dtype as numpy uint8 (this dictates the dtype of the output object).

Reply:
I am not sure I follow. I think what I was trying to say is that `actual_img` and `expected_img` are both TF tensors, so it seem natural that we should convert both (or not cover either) to numpy ndarrays before the assertion in line 108, i.e. should it not be

```
self.assertTrue(onp.allclose(actual_img.numpy(), expected_img.numpy()))
```

If it works in the current form, then my comment is just a small nit an can be ignored.

Reply:
Thanks for catching this alexey!

I think in the initial comments when I was using tensors for inputs (instead of ndarray or devicearray), I needed to extract the numpy ndarrays, that doesn't seem to be necessary now. Resolved in other occurrences as well!